crates/tab_switcher/src/tab_switcher.rs

```diff
@@ -266,14 +317,50 @@ impl PickerDelegate for TabSwitcherDelegate {
 
         let label = tab_match.item.tab_content(Some(tab_match.detail), true, cx);
         let indicator = render_item_indicator(tab_match.item.boxed_clone(), cx);
+        let indicator_color = if let Some(ref indicator) = indicator {
+            indicator.color
+        } else {
+            Color::default()
+        };
+        let indicator = h_flex()
+            .flex_shrink_0()
+            .children(indicator)
+            .child(div().w_2())
+            .into_any_element();
+        let close_button = div()
+            // We need this on_mouse_up here instead of on_click on the close
+            // button because Picker intercepts the same events and handles them
+            // as click's on list items.
+            // See the same handler in Picker for more details.
+            .on_mouse_up(
+                MouseButton::Right,
+                cx.listener(move |picker, _: &MouseUpEvent, cx| {
+                    cx.stop_propagation();
+                    picker.delegate.close_item_at(ix, cx);
+                }),
+            )
+            .child(
+                IconButton::new("close_tab", IconName::Close)
+                    .icon_size(IconSize::Small)
+                    .icon_color(indicator_color)
+                    .tooltip(|cx| Tooltip::text("Close", cx)),
+            )
+            .into_any_element();
 
         Some(
             ListItem::new(ix)
                 .spacing(ListItemSpacing::Sparse)
                 .inset(true)
                 .selected(selected)
                 .child(h_flex().w_full().child(label))
-                .children(indicator),
+                .when(true, |el| {
```

Comment:
This `when(true` smells a bit. Could we perhaps use `map` or `when_else` there? `map` in particular should be a bit less verbose.

Reply:
Oh, sure, that looks bad.

Reply:
Fixed